sup3r/preprocessing/data_handling/dual_data_handling.py

```diff
@@ -0,0 +1,614 @@
+            shape=self.lr_requested_shape, fill_value=np.nan, dtype=np.float32
+        )
+
+        logger.info(
+            f'Loading cache with requested_shape={self.lr_requested_shape}.'
+        )
+        self._load_cached_data(
+            regridded_data,
+            self.regrid_cache_files,
+            self.features,
+            max_workers=self.hr_dh.load_workers,
+        )
+
+        self.lr_data = regridded_data
+
+    def load_cached_data(self):
+        """Load regridded low_res and high_res cache data"""
+        self.load_lr_cached_data()
+        self._shape_check()
+        self._val_split_check()
+
+    def check_clear_data(self):
+        """Check if data was cached and free memory if load_cached is False"""
+        if self.regrid_cache_pattern is not None and not self.load_cached:
+            self.lr_data = None
+            self.lr_val_data = None
+            self.hr_dh.check_clear_data()
+
+    def get_lr_data(self):
+        """Check if era data is cached. If not then extract data and regrid.
+        Save to cache if cache pattern provided."""
+
+        if self.try_load:
+            self.load_lr_cached_data()
+        else:
+            regridded_data = self.regrid_lr_data()
+
+            if self.regrid_cache_pattern is not None:
+                logger.info(
+                    'Caching low resolution data with '
+                    f'shape={regridded_data.shape}.'
+                )
+                self._cache_data(
+                    regridded_data,
+                    features=self.features,
+                    cache_file_paths=self.regrid_cache_files,
+                    overwrite=self.overwrite_regrid_cache,
+                )
+        self.lr_data = regridded_data
+
+    def get_regridder(self):
+        """Get regridder object"""
+        input_meta = pd.DataFrame()
+        input_meta['latitude'] = self.lr_dh.lat_lon[..., 0].flatten()
+        input_meta['longitude'] = self.lr_dh.lat_lon[..., 1].flatten()
+        target_meta = pd.DataFrame()
+        target_meta['latitude'] = self.lr_lat_lon[..., 0].flatten()
+        target_meta['longitude'] = self.lr_lat_lon[..., 1].flatten()
+        return Regridder(
+            input_meta, target_meta, max_workers=self.regrid_workers
+        )
+
+    def regrid_lr_data(self):
+        """Regrid low_res data for all requested features
+
+        Returns
+        -------
+        out : ndarray
+            Array of regridded low_res data with all features
+            (spatial_1, spatial_2, temporal, n_features)
+        """
+        logger.info('Regridding low resolution feature data.')
+        regridder = self.get_regridder()
+
+        return np.concatenate(
+            [
+                regridder(self.lr_input_data[..., i]).reshape(
+                    self.lr_required_shape
+                )[..., np.newaxis]
+                for i in range(len(self.features))
+            ],
+            axis=-1,
+        )
+
+    def get_next(self):
```

Comment:
This method looks clean and well organized, nice job (except for the line break style that i'm not a fan of :))